src/libcore/iter/traits/iterator.rs

```diff
@@ -2572,13 +2572,13 @@ pub trait Iterator {
     /// ```
     #[inline]
     #[unstable(feature = "is_sorted", reason = "new API", issue = "53485")]
-    fn is_sorted_by_key<F, K>(self, mut f: F) -> bool
+    fn is_sorted_by_key<F, K>(self, f: F) -> bool
     where
         Self: Sized,
-        F: FnMut(&Self::Item) -> K,
+        F: FnMut(Self::Item) -> K,
```

Comment:
Can you make a writeup to convince me that this change is desirable?  I can see an argument that this is better, but I note that the other `is_sorted_by_key` and `sort_by_key` functions take references.

Reply:
Note that the closures of the slice methods `is_sorted_by_key` and `sort_by_key` take a `&T` which is the `Item` type of `slice::Iter`, not a reference to it. So this could be considered consistent with that: `slice.is_sorted_by_key(...)` and `slice.iter().is_sorted_by_key(...)` now take the same closure type (instead of the latter's closure taking a double reference).

I didn't have any particular use case in mind that requires having ownership of the values, I just figured it may as well pass the elements by value if `is_sorted_by_key` doesn't need them anymore after that. And it is a clear ergonomic win for iterators that already iterate over references.

Is that sufficient?

Reply:
Honestly with this change the source becomes so concise that I wonder why this function is even a useful addition.  Like why would I write `iter.is_sorted_by_key(|x| x.len())` rather than `iter.map(|x| x.len()).is_sorted()`?  Is this just here to make the intent clearer?

Reply:
Hmm, I like the analogy that it's not that the slice versions are over references but that they're over the item of the slice's iterator :+1: